sklearn/ensemble/_hist_gradient_boosting/_loss.pyx

```diff
@@ -201,13 +205,13 @@ cdef inline void _compute_softmax(Y_DTYPE_C [:, ::1] p, const int i) nogil:
             max_value = p[i, k]
 
     for k in range(prediction_dim):
-        p[i, k] = exp(p[i, k] - max_value)
+        p[i, k] = expf(p[i, k] - max_value)
         sum_exps += p[i, k]
 
     for k in range(prediction_dim):
         p[i, k] /= sum_exps
 
 
-cdef inline Y_DTYPE_C _cexpit(const Y_DTYPE_C x) nogil:
+cdef inline G_H_DTYPE_C _cexpit(const G_H_DTYPE_C x) nogil:
```

Comment:
this is now expecting a float but what is passed to _cexpit is still a double

Reply:
If I understand C correctly, in particular type casting, the argument of `_cexpit` will be implicitly converted to the type as specified in function definition, which is now float32.